swarm/network/stream/stream.go

```diff
@@ -474,7 +474,7 @@ type server struct {
 // setNextBatch adjusts passed interval based on session index and whether
```

Comment:
@justelad I think we should extract this logic and have it unit tested. Basically the functionality which adjusts the input intervals and uses the adjusted intervals is not clear and not tests, which is weird, as there are very little input parameters to it:
1. from
2. to
3. sessionIndex
4. stream type (live or history)

This should have been unit tested, at least for the purpose of documentation.

Reply:
this function should not even exist IMO. not sure why we would ever want to override a value. e.g. when syncing is live and from is `0`... but i think that with the first offered hashes msg the client already gets the actual session index from the server with the `from` range specified on the message. a lot of the code here is not needed, some of it also should return errors directly (which it doesn't)